pkg/autoupdate/autoupdate.go

```diff
@@ -198,14 +233,14 @@ func AutoUpdate(runtime *libpod.Runtime, options Options) ([]string, []error) {
 
 // imageContainersMap generates a map[image ID] -> [containers using the image]
 // of all containers with a valid auto-update policy.
-func imageContainersMap(runtime *libpod.Runtime) (map[string][]*libpod.Container, []error) {
+func imageContainersMap(runtime *libpod.Runtime) (map[string][]PolicyContainer, []error) {
```

Comment:
Can you extend the comment of this function? I think it'll be helpful for future readers to understand the reasoning behind this structure:

* `policyContainer` to bundle a policy with a container
* Mapping from image to a slice of policy containers since it's possible that multiple containers refer to the same image but with different policies.

Thinking about the upper points. If the slice has `[policy A, policy B, policy A]` ... I think it would make sense to further bundle the items together such that we have to run a given policy only _once_ instead of for each item individually.

A `PolicyContainer` could look as follows:
```golang
type policyMapper struct {
   map[Policy][]*libpod.Container
}
```

Now, when executing the policies we have direct mapping `image -> policy -> containers using this policy`.